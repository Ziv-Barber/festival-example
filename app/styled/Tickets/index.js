// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Other components that we need:
import { ItemsList } from '../../components/ItemsList'
import TicketsBox from '../TicketsBox'

// Content that we need:
import ticketsList from '../../content/tickets'

// Create the list of items for the Lineup component:
const ticketsItems = ticketsList.map((value) => ({
  key: value.key,
  component: TicketsBox,
  compProps: {
    name: value.name,
    image: value.image,
    price: value.price,
    desc: value.desc
  }
  // gridProps
}))

/**
 * Tickets react component.
 */
export default styled(ItemsList).attrs({
  items: ticketsItems,
  caption: 'Buy Tickets',
  captionProps: '',
  captionVariant: '',
  containerId: '',
  gridRootDefProps: {
    spacing: 8,
    justify: 'space-around',
    alignItems: 'center'
  },
  gridDefProps: {
    xs: 12,
    sm: 4,
    md: 3,
    lg: 2,
    xl: 1
  }
})`
  margin: 0 !important;
  margin-bottom: -1px !important;
  padding: ${(props) => props.theme.spacing.unit}px 0 !important;
  width: 100% !important;
  background: linear-gradient(to bottom, #99c2ff, #4d94ff);
  color: white;

  & .listHeader {
    padding: ${(props) => props.theme.spacing.unit}px !important;
    width: 100% !important;
    text-shadow: 2px 2px 5px #fff !important;
    text-align: center;
    color: #003366;
  }

  & .listGrid {
    margin: 0 !important;
    width: 100% !important;
    padding: ${(props) => props.theme.spacing.unit}px !important;
  }

  & .listItem {
    margin: ${(props) => props.theme.spacing.unit}px !important;
    z-index: 1050;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  & .card {
    border: 2px solid #fff;
    min-width: 254px;
    max-width: 254px;
    background-image: none;
    // background-color: transparent;
    background-color: #001433;
    padding: 0 !important;
  }

  & .cardImage {
    width: 254px;
    border: 2px solid #fff;
  }

  & .cardContent {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${(props) => props.theme.spacing.unit}px !important;
  }

  & .cardActions {
    flex-direction: column;
    padding: 0 !important;
  }

  & .cardDesc {
    text-align: center;
    margin: 0 !important;
    padding: ${(props) => props.theme.spacing.unit}px !important;
  }

  & .cardButton {
    padding-bottom: ${(props) => props.theme.spacing.unit * 2}px !important;
  }

  & h2 {
    color: #fff !important;
  }
`
