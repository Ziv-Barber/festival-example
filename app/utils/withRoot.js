// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
