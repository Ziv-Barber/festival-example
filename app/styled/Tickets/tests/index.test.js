// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import Tickets from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {Tickets} */
describe('<Tickets />', () => {
  afterEach(cleanup)

  /** @test {Tickets} */
  it('should render', () => {
    const App = withRoot(Tickets)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
