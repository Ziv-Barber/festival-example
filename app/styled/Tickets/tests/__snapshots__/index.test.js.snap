// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tickets /> should render 1`] = `
<div>
  <div
    class="MuiPaper-root-1 MuiPaper-elevation0-3 Tickets-q4qtd0-0 fyrttE"
  >
    <h3
      class="MuiTypography-root-28 MuiTypography-h3-42 listHeader"
    >
      Buy Tickets
    </h3>
    <div
      class="MuiGrid-container-64 MuiGrid-spacing-xs-8-86 MuiGrid-align-items-xs-center-72 MuiGrid-justify-xs-space-around-84 listGrid"
    >
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-12-104 MuiGrid-grid-sm-4-110 MuiGrid-grid-md-3-123 MuiGrid-grid-lg-2-136 MuiGrid-grid-xl-1-149 listItem"
      >
        <div
          class="MuiPaper-root-1 MuiPaper-elevation0-3 MuiCard-root-161 card"
        >
          <div
            class="MuiCardContent-root-162 cardContent"
          >
            <h2
              class="MuiTypography-root-28 MuiTypography-h6-45"
            >
              First 1000 tickets
               
              <strong>
                100€
              </strong>
            </h2>
          </div>
          <div
            class="MuiCardActions-root-163 cardActions"
          >
            <img
              class="cardImage MuiCardActions-action-165"
              src="IMAGE_MOCK"
            />
            <p
              class="MuiTypography-root-28 MuiTypography-body1-37 cardDesc MuiCardActions-action-165"
            >
              Reducted price
            </p>
            <div
              class="cardButton MuiCardActions-action-165"
            >
              <button
                class="MuiButtonBase-root-192 MuiButton-root-166 MuiButton-contained-177 MuiButton-containedPrimary-178 MuiButton-raised-180 MuiButton-raisedPrimary-181"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-167"
                >
                  <a
                    href="https://www.myticket.de/"
                  >
                    Buy now!
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root-195"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-12-104 MuiGrid-grid-sm-4-110 MuiGrid-grid-md-3-123 MuiGrid-grid-lg-2-136 MuiGrid-grid-xl-1-149 listItem"
      >
        <div
          class="MuiPaper-root-1 MuiPaper-elevation0-3 MuiCard-root-161 card"
        >
          <div
            class="MuiCardContent-root-162 cardContent"
          >
            <h2
              class="MuiTypography-root-28 MuiTypography-h6-45"
            >
              Normal tickets
               
              <strong>
                150€
              </strong>
            </h2>
          </div>
          <div
            class="MuiCardActions-root-163 cardActions"
          >
            <img
              class="cardImage MuiCardActions-action-165"
              src="IMAGE_MOCK"
            />
            <p
              class="MuiTypography-root-28 MuiTypography-body1-37 cardDesc MuiCardActions-action-165"
            >
              Normal tickets
            </p>
            <div
              class="cardButton MuiCardActions-action-165"
            >
              <button
                class="MuiButtonBase-root-192 MuiButton-root-166 MuiButton-contained-177 MuiButton-containedPrimary-178 MuiButton-raised-180 MuiButton-raisedPrimary-181"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-167"
                >
                  <a
                    href="https://www.myticket.de/"
                  >
                    Buy now!
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root-195"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-12-104 MuiGrid-grid-sm-4-110 MuiGrid-grid-md-3-123 MuiGrid-grid-lg-2-136 MuiGrid-grid-xl-1-149 listItem"
      >
        <div
          class="MuiPaper-root-1 MuiPaper-elevation0-3 MuiCard-root-161 card"
        >
          <div
            class="MuiCardContent-root-162 cardContent"
          >
            <h2
              class="MuiTypography-root-28 MuiTypography-h6-45"
            >
              V.I.P. tickets
               
              <strong>
                300€
              </strong>
            </h2>
          </div>
          <div
            class="MuiCardActions-root-163 cardActions"
          >
            <img
              class="cardImage MuiCardActions-action-165"
              src="IMAGE_MOCK"
            />
            <p
              class="MuiTypography-root-28 MuiTypography-body1-37 cardDesc MuiCardActions-action-165"
            >
              V.I.P. tickets
            </p>
            <div
              class="cardButton MuiCardActions-action-165"
            >
              <button
                class="MuiButtonBase-root-192 MuiButton-root-166 MuiButton-contained-177 MuiButton-containedPrimary-178 MuiButton-raised-180 MuiButton-raisedPrimary-181"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-167"
                >
                  <a
                    href="https://www.myticket.de/"
                  >
                    Buy now!
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root-195"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
